src/shared/dissect-image.c

```diff
@@ -2796,29 +2796,29 @@ int dissected_image_acquire_metadata(DissectedImage *m, DissectImageFlags extra_
         return r;
 }
 
-int dissect_image_and_warn(
-                int fd,
+int dissect_loop_device_and_warn(
                 const char *name,
+                const LoopDevice *loop,
                 const VeritySettings *verity,
                 const MountOptions *mount_options,
-                uint64_t diskseq,
-                uint64_t uevent_seqnum_not_before,
-                usec_t timestamp_not_before,
                 DissectImageFlags flags,
                 DissectedImage **ret) {
 
         _cleanup_free_ char *buffer = NULL;
         int r;
 
+        assert(loop);
+        assert(loop->fd >= 0);
+
         if (!name) {
-                r = fd_get_path(fd, &buffer);
+                r = fd_get_path(loop->fd, &buffer);
```

Comment:
hmm, if you have a loopback device, you could just use `loop->node` instead. Except thta there's one codepath right now where `loop->node` is not initialized, but we could fix that. i.e. in loop_device_make_internal() in the `if (S_ISBLK(st.st_mode))` branch at the top, in case we are not talking about a loopback block device, but another one. We could add a call to devpath_from_devnum() there, to get a pretty path from the stat data...

If you add that we would give the guarantee that `loop->node` is always valid, and thus you can drop the fd_get_path() stuff here